### DISCUSSION TITLE
Implementing Print Preview without background browser screen

### DISCUSSION THREAD

**Discussion by sudhirsharmat** [cefsharp/CefSharp#4883] [Ideas]
Hello Team,

I am planning to use CefSharp in one of my projects for print preview functionality. I have an existing custom document viewer, and I would like to hide the browser window, displaying only the print preview screen directly.

Could you please guide me on how to achieve this?

Thank you!

**Comment by chacham2** [comment]
So, you don't want to show the browser at all, or do you want to make the browser not visible during the duration or the print preview?

If the browser is on a separate form, you can control the visibility of that form. For example, in VB:

`New Form With {.Height = 768, .Width = 1024, .Visible = False, .ShowIcon = False}`

**Comment by sudhirsharmat** [comment]
Hi @chacham2 ,

Thank for your reply. My application has pdf viewing capability. So, pdf is already viewed to user. I have added Print and Print Preview feature in File menu. Now when user clicks on Print Preview, I want to directly show the user the print preview, without the original browser window. My application is in C#. Please suggest how to do it. 

Thank you!

**Reply by chacham2** [reply]
>show the user the print preview, without the original browser window

What is the "original browser window"?

**Reply by sudhirsharmat** [reply]
When we use CefSharp to load a PDF, it opens in the default browser window, which includes options to print or save the file. From our code, we trigger the print preview, which opens a pop-up with the PDF's print preview. However, since the PDF is already displayed in our custom viewer, we want to hide the default browser window and only show the print preview dialog.

**Reply by chacham2** [reply]
So, you want to be able to hide the browser itself? One way to hide it is to hide the browser's parent control.

**Comment by amaitland** [comment]
There's no built in functionality for this. The browser might not like being hidden when trying to show print preview.

There's no callback to notify when the print dialog has closed.
